test/integration/crud/maxTimeMS.test.ts

```diff
@@ -0,0 +1,268 @@
+        });
+      });
+    }
+  });
+
+  it('should correctly fail with maxTimeMS error', {
+    // Add a tag that our runner can trigger on
+    // in this case we are setting that node needs to be higher than 0.10.X to run
+    metadata: {
+      requires: {
+        topology: ['single', 'replicaset']
+      }
+    },
+
+    test: function (done) {
+      const configuration = this.configuration;
+      const client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
+      client.connect(function () {
+        const db = client.db(configuration.db);
+        const col = db.collection('max_time_ms_5');
+
+        // Insert a couple of docs
+        const docs_1 = [{ agg_pipe: 10 }];
+
+        // Simple insert
+        col.insertMany(docs_1, { writeConcern: { w: 1 } }, function (err) {
+          expect(err).to.not.exist;
+
+          db.admin().command(
+            { configureFailPoint: 'maxTimeAlwaysTimeOut', mode: 'alwaysOn' },
+            function (err, result) {
+              expect(err).to.not.exist;
+              test.equal(1, result?.ok);
+
+              col
+                .find({})
+                .maxTimeMS(10)
+                .toArray(function (err) {
+                  test.ok(err != null);
+
+                  db.admin().command(
+                    { configureFailPoint: 'maxTimeAlwaysTimeOut', mode: 'off' },
+                    function (err, result) {
+                      expect(err).to.not.exist;
+                      test.equal(1, result?.ok);
+                      client.close(done);
+                    }
+                  );
+                });
+            }
+          );
+        });
+      });
+    }
+  });
+
+  describe('awaitData, tailable, maxTimeMS, and maxAwaitTimeMS on cursors', () => {
+    const insertedDocs = [{ _id: 1 }];
+    let client: MongoClient;
+    let cappedCollection: Collection<{ _id: number }>;
+    let cursor: FindCursor<{ _id: number }>;
+    let events: CommandStartedEvent[];
+
+    beforeEach(async function () {
+      client = this.configuration.newClient({ monitorCommands: true });
+      await client
+        .db()
+        .dropCollection('cappedAt3')
+        .catch(() => null);
+      await sleep(100); // Need to make sure it has had time to drop entirely
+      cappedCollection = await client
+        .db()
+        .createCollection('cappedAt3', { capped: true, size: 4096, max: 3 });
+      cappedCollection.insertMany(insertedDocs);
+
+      events = [];
+      client.on('commandStarted', event =>
+        ['getMore', 'find'].includes(event.commandName) ? events.push(event) : null
+      );
+    });
+
+    afterEach(async function () {
+      events = [];
+      await cursor?.close();
+      await client?.close();
+    });
+
+    const tailableValues = [true, false, undefined];
+    const awaitDataValues = [true, false, undefined];
+    const maxTimeMSValues = [100, 0, undefined];
+    const maxAwaitTimeMSValues = [100, 0, undefined];
```

Comment:
How were these values chosen?  Is it a >0, 0 and undefined value?  

Should we also be testing with negative values?

Reply:
I don't think we need negative tests here, MongoDB will error. I tested zero because its a falsey case, and 100 is just a postive value